### DISCUSSION TITLE
Solutions for auto-import in VS Code?

### DISCUSSION THREAD

**Discussion by zenzen-sol** [shadcn-ui/ui#3619] [Q&A]
VS Code wants to pull auto-imports directly from the `/@radix-ui/react-*` package, but I want auto-imports to pull from my local shadcn-ui components. That's pretty easy to implement in VS Code's settings:

```bash
  ...,
  "typescript.preferences.autoImportFileExcludePatterns": [
    "**/@radix-ui/react-*",
  ],
  ...
```

However, I also use `@radix-ui/react-icons` in this project. The file exclusion glob pattern above will prevent the icons from being auto-imported.

I expected that `**/@radix-ui/react-!(icons)` would match everything in `@radix-ui/react-*` except `@radix-ui/react-icons`, but that is not the result. Instead, none of the Radix UI imports work after changing the VS Code settings to exclude `**/@radix-ui/react-!(icons)`.

I can't be the only person with this issue. How have others solved it?

**Comment by SerenModz21** [comment]
`!(icons)` is not a valid regex. It would be `@radix-ui/react-(?!icons)` as shown here: https://regexr.com/8e6tk
